src/Impostor.Server/Events/Game/BeforeGameCreatedEvent.cs

```diff
@@ -0,0 +1,50 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Impostor.Api.Events;
+using Impostor.Api.Innersloth;
+using Impostor.Server.Net.Redirector;
+using Impostor.Server.Net.State;
+
+namespace Impostor.Server.Events
+{
+    public class BeforeGameCreatedEvent : IBeforeGameCreatedEvent
+    {
+        private readonly ICollection<int> _gameKeys;
+
+        public BeforeGameCreatedEvent(ICollection<int> gameKeys)
+        {
+            _gameKeys = gameKeys;
+        }
+
+        public string? GameCode { get; private set; }
+
+        public bool TryToSetCode(string gameCode)
```

Comment:
just use property setter with backing field

Reply:
> The point of `TryToSetCode` is to return true if the code is available. That way, if the first code you want already exists, you can use another one, and so on. That would be impossible with a property setter.

Reply:
It still could be a property, just throw a exception that you can catch, no?

Reply:
So the plugin should catch the exception ? I find it a little dangerous, as unhandled exception would completely prevent the room creation. I know you could say that it would be "the plugin's fault" but still.

Reply:
It wouldn't actually, exception would be thrown inside event handling - so just error would be printed out

Reply:
Well ok then.